runner/engine.go

```diff
@@ -437,7 +440,7 @@ func (e *Engine) runBin() error {
 			default:
 			}
 		}()
-		<-e.binStopCh
+		<-e.binStopChMap[cmd.Process.Pid]
```

Comment:
need check nil?
bcs when we read from a nil channel it will block forever.

Reply:
I'll look into this. I think I want to make sure our locking ensures we never attempt to release before we start listening so the channel is never nil. 

I think I'll pass in the channel to the function here instead of looking it up on demand to protect against it already having been messaged.

Reply:
I did a little refactoring that should prevent reading from a nil channel.

Reply:
I dont think we need a map. just use close() to broadcast.